reactor-core/src/test/java/reactor/core/scheduler/SchedulersTest.java

```diff
@@ -1227,6 +1227,8 @@ public void testDirectScheduleZeroPeriodicallyCancelsSchedulerTask() throws Exce
 			latch.await();
 
 			disposable.dispose();
+			//avoid race of checking the status of futures vs cancelling said futures
+			Thread.sleep(10);
```

Comment:
sorry for singing the same song again, but IMO Awaitility here would serve better and not add a delay if it resolves immediately, or retry until true 😊